### PR TITLE
Update Email Alert API documentation to reflect system changes

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/G35ncFfw/645-simplify-configuration-for-email-sending

This updates a number of documents to reflect recent changes to the Email Alert API system which were introduced in https://github.com/alphagov/email-alert-api/pull/1489.

These changes are mostly done as a quick-pass of these docs as we may manage to change the system further before the weeks out. 

This is recommended to be reviewed commit by commit.

### CODE REVIEW

File: source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md [hunk 1]
```diff
@@ -6,32 +6,28 @@ layout: manual_layout
...
+1. Add your email address to the `govuk_notify_recipients` list in
+   [hieradata_aws/common.yml][].
+2. Create a branch with your changes and push them to GitHub. Deploy the
+   changes by running the [deploy-puppet][] job for the environment you're
+   testing on, providing your branch name instead of a release tag.
+3. Once these changes have been deployed, and the environment variable
+   `GOVUK_NOTIFY_RECIPIENTS` is populated with your address, you can test that
+   you can receive emails by running the [Send a test email][] support task.
 
 ## Testing digest emails
```
Comment (benthorner) [inline source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md:28]: Possibly out-of-scope: I think we should just delete this, as it's just a poor substitute for how the digest process actually works, and it's not something we've needed to do.

File: source/manual/email-troubleshooting.html.md [hunk 2]
```diff
@@ -31,8 +31,6 @@ will take you to a form where you can search by email address.
 
 A courtesy copy of every email sent from integration, staging and production
```
Comment (benthorner) [inline source/manual/email-troubleshooting.html.md:33]: Out-of-scope: we should kill this doc (it's too general to be useful) and migrate it's content elsewhere:

- The link in the first section is in other docs.

- The second section appears to be covered [here](https://docs.publishing.service.gov.uk/manual/govuk-notify.html).

- The third section is much more relevant [here](https://docs.publishing.service.gov.uk/manual/email-notifications-how-they-work.html).

File: source/manual/load-test-email-alert-api.html.md [hunk 3]
```diff
@@ -1,73 +1,27 @@
...
+bottlenecks.
 
-Before you begin, you should post in the #2nd-line channel that you’re about to do some load testing and that as part of that you’ll be disabling puppet runs on Staging for email-alert-api.
+To perform a load test you will need:
 
-> Rake tasks for load testing can be found [here](https://github.com/alphagov/email-alert-api/blob/master/lib/tasks/load_testing.rake).
+- A mechanism to artificially create a quantity of work for Email Alert API to
+  do - we previously had a number of [rake tasks][] to allow this;
+- An approach to simulate the delay of an actual request to Notify - we
+  previously had a [DelayProvider][] class for this.
```
Comment (benthorner) [inline source/manual/load-test-email-alert-api.html.md:19]:
```suggestion
  previously used `Kernel.sleep 0.1` to simulate this.
```

I think it's more confusing to link to code that no longer exists, as the reader would have to work out what replaced it.

**Review by benthorner — APPROVED** [review]
Looks good 📚. I've added a few minor comments about other changes we could make, but this is a net improvement on its own, and removes all the inconsistencies around restricted sending in Integration and Staging.

File: source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md [hunk 4]
```diff
@@ -6,32 +6,28 @@ layout: manual_layout
...
-
-3. Once these changes have been deployed and the environment variable `EMAIL_ADDRESS_OVERRIDE_WHITELIST` is populated with your address you can test that you can receive emails by running the [support:send_test_email[name@example.com]](https://deploy.integration.publishing.service.gov.uk/job/run-rake-task/parambuild/?TARGET_APPLICATION=email-alert-api&MACHINE_CLASS=email_alert_api&RAKE_TASK=support:send_test_email[your.name@digital.cabinet-office.gov.uk]) rake task.
+1. Add your email address to the `govuk_notify_recipients` list in
+   [hieradata_aws/common.yml][].
+2. Create a branch with your changes and push them to GitHub. Deploy the
+   changes by running the [deploy-puppet][] job for the environment you're
+   testing on, providing your branch name instead of a release tag.
+3. Once these changes have been deployed, and the environment variable
+   `GOVUK_NOTIFY_RECIPIENTS` is populated with your address, you can test that
```
[the remark this answers is not in the document]
Comment (kevindew) [reply source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md:25]: I think this is ok to not mention given that puppet will automatically restart the app for you when the env var changes.

File: source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md (hunk 1 above)
Comment (kevindew) [reply source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md:28]: Yeah, I'm not convinced by this section. I'll plan to chop it by the end of the week if I manage to get the groups sorted.

File: source/manual/email-troubleshooting.html.md (hunk 2 above)
Comment (kevindew) [reply source/manual/email-troubleshooting.html.md:33]: Sure, I'll see if I can take that on later in the week.

File: source/manual/load-test-email-alert-api.html.md (hunk 3 above)
Comment (kevindew) [reply source/manual/load-test-email-alert-api.html.md]: Thanks, I've applied some very similar.

File: source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (benthorner) [reply source/manual/receiving-emails-from-email-alert-api-in-integration-and-staging.html.md:25]: Oh yeah, I forgot it does do that.

**Pull request merged**